src/debugger/Debugger.js

```diff
@@ -110,38 +113,75 @@ export class DebugServer {
   // Process a command from a debugger. Returns whether Prepack should unblock
   // if it is blocked
   processDebuggerCommand(request: DebuggerRequest) {
+    let requestID = request.id;
     let command = request.command;
     let args = request.arguments;
     switch (command) {
       case DebugMessage.BREAKPOINT_ADD_COMMAND:
         invariant(args.kind === "breakpoint");
         this._breakpoints.addBreakpoint(args.filePath, args.line, args.column);
-        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_ADD_ACKNOWLEDGE, args);
+        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_ADD_ACKNOWLEDGE, requestID, args);
         break;
       case DebugMessage.BREAKPOINT_REMOVE_COMMAND:
         invariant(args.kind === "breakpoint");
         this._breakpoints.removeBreakpoint(args.filePath, args.line, args.column);
-        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_REMOVE_ACKNOWLEDGE, args);
+        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_REMOVE_ACKNOWLEDGE, requestID, args);
         break;
       case DebugMessage.BREAKPOINT_ENABLE_COMMAND:
         invariant(args.kind === "breakpoint");
         this._breakpoints.enableBreakpoint(args.filePath, args.line, args.column);
-        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_ENABLE_ACKNOWLEDGE, args);
+        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_ENABLE_ACKNOWLEDGE, requestID, args);
         break;
       case DebugMessage.BREAKPOINT_DISABLE_COMMAND:
         invariant(args.kind === "breakpoint");
         this._breakpoints.disableBreakpoint(args.filePath, args.line, args.column);
-        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_DISABLE_ACKNOWLEDGE, args);
+        this._channel.sendBreakpointAcknowledge(DebugMessage.BREAKPOINT_DISABLE_ACKNOWLEDGE, requestID, args);
         break;
       case DebugMessage.PREPACK_RUN_COMMAND:
         invariant(args.kind === "run");
         return true;
+      case DebugMessage.STACKFRAMES_COMMAND:
+        invariant(args.kind === "stackframe");
+        this.processStackframesCommand(requestID, args);
+        break;
       default:
         throw new DebuggerError("Invalid command", "Invalid command from adapter: " + command);
     }
     return false;
   }
 
+  processStackframesCommand(requestID: number, args: StackframeArguments) {
+    let frameInfos: Array<Stackframe> = [];
+    let loc = this._realm.currentLocation;
+
+    // iterate the stack starting from most recent context
```

Comment:
Does VSP debug protocol expect the top frame at index 0?

Reply:
The protocol does not specify, but Nuclide displays the current frame as frame 0. The index is mostly to identify the frame for other requests like scopes and variables so as long as the indices are unique.

Reply:
Then please change the comment to indicate this fact. Your current comment explains what this does not why this needs to be done.